File: Demo-LLP-MIP/Demo-LLP-MIP-Python/security/Base64.py
import base64
import codecs

class Base64:
    @staticmethod
    def get_base64(s):
        if s is None:
            return None
        return Base64.get_base64_from_bytes(s.encode('utf-8'))

    @staticmethod
    def get_base64_from_bytes(b):
        rb = base64.b64encode(b)
        if rb is None:
            return None
        return rb.decode('utf-8')

    @staticmethod
    def get_from_base64(s):
        if s is None:
            return None
        b = Base64.get_bytes_from_base64(s)
        if b is None:
            return None
        return b.decode('utf-8')

    @staticmethod
    def get_bytes_from_base64(s):
        if s is None:
            return None
        b = base64.b64decode(s.encode('utf-8'))
        return b

File: Demo-LLP-MIP/Demo-LLP-MIP-PHP/security/Base64.php
<?php
class Base64 {
    public static function get_base64($s) {
        if ($s === null) {
            return null;
        }
        return self::get_base64_from_bytes(mb_convert_encoding($s, 'UTF-8'));
    }

    public static function get_base64_from_bytes($b) {
        $rb = base64_encode($b);
        if ($rb === null) {
            return null;
        }
        return mb_convert_encoding($rb, 'UTF-8');
    }

    public static function get_from_base64($s) {
        if ($s === null) {
            return null;
        }
        $b = self::get_bytes_from_base64($s);
        if ($b === null) {
            return null;
        }
        return mb_convert_encoding($b, 'UTF-8');
    }

    public static function get_bytes_from_base64($s) {
        if ($s === null) {
            return null;
        }
        $b = base64_decode(mb_convert_encoding($s, 'UTF-8'));
        return $b;
    }
}
?>